packages/viewer/src/modules/Converter.js

```diff
@@ -259,34 +297,50 @@ export default class Coverter {
     
     delete obj.segments
     delete obj.speckle_type
+    delete obj.bbox
 
     return new ObjectWrapper( geometry , obj, 'line' )
   }
   
   async CurveToBufferGeometry( object ) {
+    
     let obj = {}
     Object.assign( obj,object )
+
     delete object.value
     delete object.speckle_type
     delete object.displayValue
-    
-    obj.points = await this.dechunk( obj.points )
-    obj.weights = await this.dechunk( obj.weights )
-    obj.knots = await this.dechunk( obj.knots )
+    delete object.bbox
+
+    obj.weights = await this.dechunk( object.weights )
+    obj.knots = await this.dechunk( object.knots )
+    obj.points = await this.dechunk( object.points )
 
     try {
+      console.log( 'Curve to buffer', object, obj )
+
+      //TODO: This should be removed when we improve the nurbs curve's in THREE.js (or make our own).
+      throw new Error( 'Skipping nurbs for displayValue due to lack of support in THREE.js of some nurbs types' )
+      
       let conversionFactor = getConversionFactor( obj.units )
       
       // Convert points+weights to Vector4
       const points = []
       for ( let i = 0; i < obj.points.length; i+=3 ) {
-        points.push( new THREE.Vector4( obj.points[ i ]* conversionFactor,obj.points[i+1]* conversionFactor,obj.points[i+2] * conversionFactor, obj.weights[i/3] * conversionFactor ) )
+        points.push( new THREE.Vector4( obj.points[ i ]* conversionFactor,obj.points[i+1]* conversionFactor,obj.points[i+2] * conversionFactor, obj.weights[i/3] ) )
       }
-      // Convert knots from rhino compact format to normal format.
-      let knots = [ obj.knots[0] ]
-      knots = knots.concat( obj.knots )
-      knots.push( knots[knots.length -1] )
-  
+
+      let knots = []
+      if ( obj.knots.length != ( obj.points.length/3 + obj.degree + 1 ) ) {
+        // Convert knots from rhino compact format to normal format.
+        let knots = [ obj.knots[0] ]
+        knots = knots.concat( obj.knots )
+        knots.push( knots[knots.length -1] )
+      }
+      else {
+        knots = obj.knots
+      }
+
       // Create the nurbs curve
       const curve = new NURBSCurve( obj.degree, knots, points, null, null )
```

Comment:
I'll remove this try catch & remove the console logs, and just comment the code out. we'll get to it when we get to it :)

Reply:
the main reason is that try catches on hot paths, when we hit the catch always, are mojo killers...